CMake/deps/SourceLocation.cmake

```diff
@@ -0,0 +1,11 @@
+include(CheckIncludeFileCXX)
+
+check_include_file_cxx("source_location" SOURCE_LOCATION_AVAILABLE)
```

Comment:
on mac this somehow finds `boost/assert/source_location.hpp` and so in Logger.h we check that builtin is available on top of this.